clients/python/src/osparc/_api_files_api.py

```diff
@@ -251,11 +251,15 @@ def _search_files(
         sha256_checksum: Optional[str] = None,
```

Comment:
Here is a thought. 
Lets use explicit sentinels to indicate that an input argument was not set
```python
class UnSet:
    ...

_unset: Final = UnSet()

def as_dict_exclude_unset(**params) -> dict[str, Any]:
    # name inspired in pydantic's model.dict(exclude_unset=True)
    return {k: v for k, v in params.items() if not isinstance(v, UnSet)}


def my_fun( must: float, nullable_optional: int | None | UnSet = _unset, string_optional: str | UnSet = _unset  ):
        kwargs = as_dict_exculde_unset(must=must, nullable_optional=nullable_optional, string_optional=string_optional)

Reply:
That's a nice idea. I like it.
But I think the issue here is the way the `openapi.json` is generated. Perhaps this will be fixed with the upgrade to pydantic v2. My point is that these params should be allowed to be `None` on the server side (https://github.com/ITISFoundation/osparc-simcore/blob/master/services/api-server/src/simcore_service_api_server/api/routes/files.py#L300). In fact that's the default value for these parameters. But somehow that is not recorded in the openapi specs (https://github.com/bisgaard-itis/osparc-simcore-clients/blob/f8be1c8418ead3809df783fa0328e92cbabbfcdc/api/openapi.json#L747). But with this wrapping they are allowed to be `None` again.

Reply:
aha, yes, nullable parameters are solved with pydnatic 2